codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexHandler.java

```diff
@@ -282,9 +280,13 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         ctx.fireExceptionCaught(cause);
     }
 
+    private static boolean isServer(ChannelHandlerContext ctx) {
+        return ctx.channel().parent() instanceof ServerChannel;
```

Comment:
This feels pretty brittle. Would this be better as a protected member of `Http2ChannelDuplexHandler` which has access to the `connection()` through the `frameCodec` field?

Reply:
we could do this but I wanted to keep the handlers as much as separated as possible ...